srml/balances/src/lib.rs

```diff
@@ -165,7 +165,12 @@ decl_storage! {
 						// <= begin it should be >= balance
 						// >= begin+length it should be <= 0
 
-						let per_block = balance / length;
+						let per_block = if length.is_zero() {
```

Comment:
If the comments above are correct it seems that entire `offset` is going to be `0`, no? Maybe we can simplify the entire vec generation somehow?

Reply:
`per_block` will be `0`, the `offset` will be set to the balance. The effect of setting a vesting schedule of length `0` is that the amount locked will never vest `locked_at(n)` will always be `self.offset.max(0) - 0`.

Reply:
Thanks for explanations!

Reply:
I'm still not entirely sure about the semantics of the `begin` parameter. Is it supposed to be the block number at which the vesting period starts? Because it doesn't seem like the balance is free to use before that period anyway, so it should always start at `0` in practice.